### PR TITLE
Update Dilithium

### PR DESCRIPTION
**Pull request by TomerHawk**
replaced parameter sets, updated benchmarks and deleted reference and pqm4 because related parameter sets changed in round 3

### CODE REVIEW

File: signatures/dilithium/dilithium-shake/impl/avx2.yaml [hunk 1]
```diff
@@ -1,5 +1,6 @@
-name: AVX2, vectorized SHAKE
+name: AVX2, using AES from OpenSSL
```
Comment (benj-zen) [inline signatures/dilithium/dilithium-shake/impl/avx2.yaml:1]: something wrong here

File: signatures/dilithium/dilithium-shake/impl/avx2.yaml [hunk 2]
```diff
@@ -1,5 +1,6 @@
...
 platform: c
 type: optimized
 hardware features:
   - avx2
+  - aes-ni
```
Comment (benj-zen) [inline signatures/dilithium/dilithium-shake/impl/avx2.yaml:6]: Don't think so

**Review by benj-zen — CHANGES_REQUESTED** [review]
Sizes and timings look ok so far. I think the security levels are not correct. We need to discuss, which values we pick.

File: signatures/dilithium/dilithium-shake/impl/avx2.yaml (hunk 1 above)
Comment (TomerHawk) [reply signatures/dilithium/dilithium-shake/impl/avx2.yaml:1]: Sorry, I think something went totally wrong

File: signatures/dilithium/dilithium-shake/impl/avx2.yaml (hunk 2 above)
Comment (TomerHawk) [reply signatures/dilithium/dilithium-shake/impl/avx2.yaml]: probably I accidentally copied impl of aes to shake version

**Review by benj-zen — APPROVED** [review]

**Pull request merged**